wp-admin/js/widgets/media-widgets.js

```diff
@@ -713,6 +713,20 @@ wp.mediaWidgets = ( function( $ ) {
 				modelProps.attachment_id = mediaFrameProps.id;
 			}
 
+			isEmbed = 0 === modelProps.attachment_id;
+
+			// Reset all video/audio sources.
+			wp.media.view.settings.embedExts.map( function( ext ) {
+
+				// Only reset when the main source changes.
+				if ( ! ( ext in modelProps ) || isEmbed ) {
+					modelProps[ ext ] = '';
+				}
+			} );
+
+			extension = mediaFrameProps.url.split( '.' ).pop();
```

Comment:
Do we need to be concerned about case here, i.e. if a file has an extension of `.MP4` instead of `.mp4` maybe call `toLowerCase()` on it just to be safe?